src/planwise/client/projects/handlers.cljs

```diff
@@ -111,16 +112,39 @@
     :facilities :facilities
     :transport :facilities-with-demand))
 
+(defn- update-tabs-state [db next-tab]
+  (let [tabs (get-in db [:current :wizard :tabs])
+        new-tabs (into {}
+                       (map (fn [[tab state]]
+                              (if (and (= state :visiting) (not= tab next-tab))
+                                [tab :visited]
+                                [tab state])) tabs))]
+    (assoc-in db [:current :wizard :tabs] new-tabs)))
+
+(defn- update-next-tab-state [db next-tab]
+  (if (= :unvisited (get-in db [:current :wizard :tabs next-tab]))
+    (assoc-in db [:current :wizard :tabs next-tab] :visiting)
+    db))
+
+(defn- update-wizard-state [db next-tab]
+  (-> db
+      (update-next-tab-state next-tab)
+      (update-tabs-state next-tab)))
+
 (register-handler
  :projects/navigate-project
  in-projects
  (fn [db [_ project-id section]]
-   (if (not= project-id (get-in db [:current :project-data :id]))
-     (dispatch [:projects/load-project project-id (section->with section)])
-     (case section
-       (:facilities :transport) (dispatch [:projects/load-facilities])
-       nil))
-   db))
+   (let [changed-project? (not= project-id (get-in db [:current :project-data :id]))
+         new-db (if changed-project?
+                  db
+                  (update-wizard-state db section))]
+     (if changed-project?
+       (dispatch [:projects/load-project project-id (section->with section)])
+       (case section
+         (:facilities :transport) (dispatch [:projects/load-facilities])
+         nil))
+     new-db)))
```

Comment:
It looks awkward to have two `ifs` on the same condition, right next to each other. Maybe we could merge them?